src/go/k8s/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -493,315 +484,119 @@ func setHelmLabelsAndAnnotations(object client.Object, rp *v1alpha2.Redpanda) {
+	manifests = append(manifests, asObj(redpanda.ClusterRoleBindings(dot))...)
+	manifests = append(manifests, asObj(redpanda.ClusterRoles(dot))...)
+	manifests = append(manifests, asObj(redpanda.LoadBalancerServices(dot))...)
+	manifests = append(manifests, asObj(redpanda.Secrets(dot))...)
 
-func (r *RedpandaReconciler) reconcileDelete(ctx context.Context, rp *v1alpha2.Redpanda) (ctrl.Result, error) {
-	if err := r.deleteHelmRelease(ctx, rp); err != nil {
-		return ctrl.Result{}, err
-	}
-	if controllerutil.ContainsFinalizer(rp, FinalizerKey) {
-		controllerutil.RemoveFinalizer(rp, FinalizerKey)
-		if err := r.Client.Update(ctx, rp); err != nil {
-			return ctrl.Result{}, err
-		}
-	}
-	return ctrl.Result{}, nil
+	return manifests, nil
 }
 
-func (r *RedpandaReconciler) createHelmRelease(ctx context.Context, rp *v1alpha2.Redpanda) (*helmv2beta2.HelmRelease, error) {
-	// create helmRelease resource from template
-	hRelease, err := r.createHelmReleaseFromTemplate(ctx, rp)
+func (r *RedpandaReconciler) reconcile(ctx context.Context, rp *v1alpha2.Redpanda) (*v1alpha2.Redpanda, ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx).WithName("RedpandaReconciler.reconcile")
+
+	manifests, err := r.getManifests(rp)
 	if err != nil {
-		r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityError, fmt.Sprintf("could not create helm release template: %s", err))
-		return hRelease, fmt.Errorf("could not create HelmRelease template: %w", err)
+		return nil, ctrl.Result{}, err
 	}
 
-	// create helmRelease object here
-	if err := r.Client.Create(ctx, hRelease); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityError, err.Error())
-			return hRelease, fmt.Errorf("failed to create HelmRelease '%s/%s': %w", rp.Namespace, rp.Status.HelmRelease, err)
+	for _, resource := range manifests {
+		// Nil unboxing issue
+		if reflect.ValueOf(resource).IsNil() {
+			continue
 		}
-		// we already exist, then update the status to rp
-		rp.Status.HelmRelease = rp.GetHelmReleaseName()
-	}
-
-	// we have created the resource, so we are ok to update events, and update the helmRelease name on the status object
-	r.event(rp, rp.Status.LastAttemptedRevision, v1alpha2.EventSeverityInfo, fmt.Sprintf("HelmRelease '%s/%s' created ", rp.Namespace, rp.GetHelmReleaseName()))
-	rp.Status.HelmRelease = rp.GetHelmReleaseName()
-
-	return hRelease, nil
-}
 
-func (r *RedpandaReconciler) deleteHelmRelease(ctx context.Context, rp *v1alpha2.Redpanda) error {
-	if rp.Status.HelmRelease == "" {
-		return nil
-	}
+		log.V(DebugLevel).Info("resource created", "resource", resource)
 
-	var hr helmv2beta2.HelmRelease
-	hrName := rp.Status.GetHelmRelease()
-	err := r.Client.Get(ctx, types.NamespacedName{Namespace: rp.Namespace, Name: hrName}, &hr)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			rp.Status.HelmRelease = ""
-			rp.Status.HelmRepository = ""
-			return nil
+		err := r.Create(ctx, resource)
```

Comment:
I think we'll want this to be a server side apply. Then we don't need to special case creations vs deletions.

Reply:
Agree, I will look how Andrew did it in Users controller.